### ISSUE TITLE
Question: using docker as a build tool?

### ISSUE THREAD

**Issue by ronaldtse** [tamatebako/tebako#21]
When building complex deliverables some developers like to use a clean environment using a new docker container. 

Especially in a task like this where dependencies can be complex. 

I wonder if it is appropriate to think that a developer would create a clean docker container, build and run the resulting executable, then use tebako outside the container to package that executable?

How would the developer do that? Docker has the ability to extract files/directories out of docker volumes or docker containers.

**Comment by maxirmx** [comment]
ruby-packer builds the executable itself. I believe it makes sence since the packager should know all binary objects that have been used so that such obkects are packaged also.

So, for the complex application developer provides Gemfile and tebako collects and builds all dependencies and then deploys them to packaged filesystem. It can happen outside of docker container that has been used for development.

**Comment by maxirmx** [comment]
The other option is to install tebako to docker development container and have tebako deploy solution created from local sources.

**Comment by ronaldtse** [comment]
Indeed, both scenarios would work, depending on what the user wishes for:
- docker built, tebako packages from inaide docker 
- docker built, tebako packages from outside  (but may have to start the container anyway)

Now the next question from the developer  is whether there is a way to inspect the built image to verify its contents. 

And that is possible via mounting / using libdwarfs to list out the file system tree and even extracting them into a separate folder.

**Comment by maxirmx** [comment]
> Now the next question from the developer is whether there is a way to inspect the built image to verify its contents.
> 
> And that is possible via mounting / using libdwarfs to list out the file system tree and even extracting them into a separate folder.

Correct. I will add a test to CI to check that newly created image is can be mounted and verified.
